### PR TITLE
Update dependencies to latest possible versions

### PR DESCRIPTION
**Pull request by atavism**
flutter pub upgrade --major-versions
Resolving dependencies... (1.0s)
Changed 6 constraints in pubspec.yaml:
- mime: ^1.0.6 -> ^2.0.0
- i18n_extension: ^12.0.1 -> ^13.0.2
- loader_overlay: ^2.3.2 -> ^4.0.3
- flutter_local_notifications: ^17.2.2 -> ^18.0.1
- device_info_plus: ^10.1.2 -> ^11.1.1
- ffigen: ^14.0.1 -> ^15.0.0

### CODE REVIEW

File: desktop/lib.go [hunk 1]
```diff
@@ -287,16 +287,16 @@ func redeemResellerCode(email, currency, deviceName, resellerCode *C.char) *C.ch
 		ResellerCode:   C.GoString(resellerCode),
 		Provider:       "reseller-code",
 	})
-	log.Debugf("DEBUG: redeeming reseller code response: %v", response)
-	if response.Error != "" {
-		log.Debugf("DEBUG: error while redeeming reseller code reponse is: %v", response.Error)
-		return sendError(errors.New("Error while redeeming reseller code: %v", response.Error))
-	}
```
Comment (atavism) [inline desktop/lib.go:294]: This error check needed to be moved up. Fixes the following issue:

```
Nov 11 18:23:13.844 - 0m46s DEBUG webclient: response.go:66 response body: {"error":"wrong-reseller-code","errorId":"wrong-reseller-code","status":"error"} status code 200
Nov 11 18:23:13.845 - 0m46s DEBUG lantern-client.main: lib.go:290 DEBUG: redeeming reseller code response: <nil>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x140e7efec]

goroutine 66 [running, locked to thread]:
main.redeemResellerCode(0x600001d6d420, 0x600001f7c4f0, 0x600001f72070, 0x600001d6c340)
	github.com/getlantern/lantern-client/desktop/lib.go:291 +0x24c
```

File: pubspec.yaml [hunk 2]
```diff
@@ -87,7 +87,7 @@ dependencies:
 
   # Timer & Overlay
   stop_watch_timer: ^3.2.1
-  loader_overlay: ^2.3.2
+  loader_overlay: ^4.0.3
```
Comment (jigar-f) [inline pubspec.yaml:90]: This is big change across app, We need to test this well.

File: desktop/lib.go (hunk 1 above)
Comment (jigar-f) [reply desktop/lib.go:294]: Thanks for fixing this.

**Review by jigar-f — APPROVED** [review]
LGTM, Will test this shortly.

**Comment by atavism** [comment]
@jigar-f Thanks for the review. Feel free to merge it after you've done testing if all works well

**Comment by jigar-f** [comment]
The changes looks good, merging

**Pull request merged**